src/tactics/FStar.Tactics.V2.Basic.fst

```diff
@@ -2106,7 +2106,26 @@ let dbg_refl (g:env) (msg:unit -> string) =
   then BU.print_string (msg ())
 
 let issues = list Errors.issue
-let refl_typing_builtin_wrapper (f:unit -> 'a) : tac (option 'a & issues) =
+
+let rec iter (f : 'a -> tac unit) (xs : list 'a) : tac unit =
+  match xs with
+  | [] -> ret ()
+  | x::xs ->
+    f x;!
+    iter f xs
+
+let refl_typing_guard (e:env) (g:typ) : tac unit =
+  let reason = "refl_typing_guard" in
+  proc_guard_formula "refl_typing_guard" e g None Range.dummyRange
```

Comment:
So we just process the guards according to the policy, and then they will be handled by the Tactics.Monad.run at the end as usual?

Reply:
Yes, the policy dictates whether a guard is to be solves immediately by SMT (`SMTSync`), or immediately without SMT (`Force`) or added as a goal (`Goal` / `SMT`, the latter should probably be called `SMTGoal`). Once it's in the proofstate the engine will keep track of it, and force it be discharged at some point. In Pulse, after [this PR](https://github.com/FStarLang/steel/pull/65), they would all be solved at the end.

Note that if the policy is not `Force`/`SMTSync`, then a reflection call could 'succeed' returning some unprovable goal, which I think is expected with this interface. This can always be avoided by temporarily changing the policy, as here https://github.com/FStarLang/steel/pull/65/files#diff-34e0e2aac381e650cef38feb50f4420ca101eff6b609e240a18c077ba7663300R6